token/program2/src/instruction.rs

```diff
@@ -26,17 +26,18 @@ pub enum TokenInstruction {
     /// Accounts expected by this instruction:
     ///
     ///   0. `[writable]` The mint to initialize.
-    ///   1.
-    ///      * If supply is non-zero: `[writable]` The account to hold all the newly minted tokens.
-    ///      * If supply is zero: `[]` The owner/multisignature of the mint.
-    ///   2. `[]` (optional) The owner/multisignature of the mint if supply is non-zero, if
-    ///                      present then further minting is supported.
+    ///   1. If supply is non-zero: `[writable]` The account to hold all the newly minted tokens.
```

Comment:
wdyt about removing `amount: u64` and the optional key 1?   Now with `MintTo`, it's trivial to add a `MintTo` instruction immediately following `InitializeMint`.  And then for a fixed supply, the third instruction is a `SetAuthority(AuthorityType::MintTokens, None)`.   A little more instructions but much more clear and will reduce the complexity of `InitializeMint`

Reply:
I also like this idea, simple composable instructions.  The risks of not doing these operations atomically should be well documented so its clear they should be in the same tx.

Reply:
Also, seems like this idea is a seperate effort for a different pr

Reply:
https://github.com/solana-labs/solana-program-library/issues/308